docs/source/java/flight.rst

```diff
@@ -0,0 +1,213 @@
+    }
+
+Cancellation and Timeouts
+=========================
+
+When making a call, clients can optionally provide `CallOptions`. This allows
+clients to set a timeout on calls. Also, some objects returned by client RPC calls
+expose a cancel method which allows terminating a call early.
+
+.. code-block:: Java
+
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+
+        Iterator<Result> resultIterator = tutorialFlightClient.doAction(
+                new Action("test-timeout"),
+                CallOptions.timeout(2, TimeUnit.SECONDS)
+        );
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+
+On the server side, timeouts are transparent. For cancellation, the server needs to manually poll
+``setOnCancelHandler`` or ``isCancelled`` to check if the client has cancelled the call,
+and if so, break out of any processing the server is currently doing.
+
+.. code-block:: Java
+
+    // Client
+    Location location = Location.forGrpcInsecure("0.0.0.0", 58609);
+    try(BufferAllocator allocator = new RootAllocator();
+        FlightClient tutorialFlightClient = FlightClient.builder(allocator, location).build()){
+        try(FlightStream flightStream = flightClient.getStream(new Ticket(new byte[]{}))) {
+            // ...
+            flightStream.cancel("tutorial-cancel", new Exception("Testing cancellation option!"));
+        }
+    } catch (Exception e) {
+        e.printStackTrace();
+    }
+    // Server
+    @Override
+    public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        // ...
+        listener.setOnCancelHandler(()->{
+                    // Implement logic to handle cancellation option
+                });
+    }
+
+Enabling TLS
+============
+
+TLS can be enabled when setting up a server by providing a
+certificate and key pair to ``FlightServer.Builder.useTls``.
+
+On the client side, use ``Location.forGrpcTls`` to create the Location for the client.
+
+Enabling Authentication
+=======================
+
+.. warning:: Authentication is insecure without enabling TLS.
+
+Handshake-based authentication can be enabled by implementing
+Handlers on the client and server side (i.e.: ``ServerAuthHandler``).
+There are two options available user/password and token.
+
+Authentication consists of two parts: on
+initial client connection, the server and client authentication
+implementations can perform any negotiation needed; then, on each RPC
+thereafter, the client provides a token. The client set an ID result
+of user/password through ``CallHeaders`` / ``Metadata`` these IDs is
+recovered by the server side to do validation. The server authentication
+handler validates the token and provides the identity of the
+client. This identity can be obtained from the
+``CallContext.peerIdentity``.
+
```

Comment:
As evidence of what I mean, this is mixing up how different authentication schemes work. Again, the other PR already explains the basic, handshake-based authentication - let's stick to just explaining that instead of trying to explain all three schemes at once.

Reply:
Thank you for the clarification. Updated.